cmake/cmaize/project_specification/project_specification.cmake

```
@@ -0,0 +1,235 @@
+    # 
+    # .. note::
+    #
+    #    Currently, only single-configuration generators are supported since
+    #    the ``build_type`` is based on ``CMAKE_BUILD_TYPE``. This could be
+    #    expanded to multi-configuration generators later, although the
+    #    build type will need to be determined at build time instead of
+    #    configure time. See `CMAKE_CONFIGURATION_TYPES 
+    #    <https://cmake.org/cmake/help/latest/variable/CMAKE_CONFIGURATION_TYPES.html#variable:CMAKE_CONFIGURATION_TYPES>`__
+    #    and `CMake Build Configurations 
+    #    <https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-configurations>`__
+    #    for more information on the details of making this switch.
+    #]]
+    cpp_attr(ProjectSpecification build_type)
+
+    #[[[
+    # :type: Toolchain
+    #
+    # User-specified and autopopulated toolchain file values.
+    #]]
+    cpp_attr(ProjectSpecification toolchain)
+
+    #[[[
+    # :type: cpp_map
+    #
+    # Project configure options.
+    #
+    # This is initialized to an empty map for users to fill.
+    #]]
+    cpp_attr(ProjectSpecification configure_options)
+
+    #[[[
+    # :type: cpp_map
+    #
+    # Project compile options.
+    #
+    # This is initialized to an empty map for users to fill.
+    #]]
+    cpp_attr(ProjectSpecification compile_options)
+
+    #[[[
+    # Default constructor for ProjectSpecification object with only
+    # autopopulated options available.
+    #
+    # :param self: ProjectSpecification object constructed.
+    # :type self: ProjectSpecification
+    # :returns: ``self`` will be set to the newly constructed
+    #           ``ProjectSpecification`` object.
+    # :rtype: ProjectSpecification
+    #]]
+    cpp_constructor(CTOR ProjectSpecification)
+    function("${CTOR}" self)
+
+        # Initialize the ProjectSpecification object
+        ProjectSpecification(__initialize "${self}")
+
+    endfunction()
+
+    #[[[
+    # Hash the object using the provided hashing algorithm string.
+    #
+    # Supported hashing algorithms are specified by the CMake
+    # ``string(<HASH>`` function defined `here
+    # <https://cmake.org/cmake/help/latest/command/string.html#hashing>`__.
+    #
+    # :param self: ``ProjectSpecification`` object to hash.
+    # :type self: ProjectSpecification
+    # :param return_hash: Hashed ``ProjectSpecification``
+    # :type return_hash: str
+    # :param hash_type: Hash algorithm to use
+    # :type hash_type: str
+    #
+    # :returns: Hashed ``ProjectSpecification`` object
+    # :rtype: str
+    #]]
+    cpp_member(hash ProjectSpecification str str)
+    function("${hash}" self return_hash hash_type)
+
+        cpp_serialize(self_serialized "${self}")
+
+        string("${hash_type}" "${return_hash}" "${self_serialized}")
+
+        cpp_return("${return_hash}")
+
+    endfunction()
+
+    #[[[
+    # Override to ``set_version()`` method to catch when the project version
```

```suggestion
    # Overload to ``set_version()`` method to catch when the project version
```
I think you meant overload, not override.

---

Whoops, you're right! I get those mixed up all the time.